lib/templates.php

```diff
@@ -177,6 +202,13 @@ function filter_rest_wp_template_collection_params( $query_params ) {
  * @return array Filtered $args.
  */
 function filter_rest_wp_template_query( $args, $request ) {
+	// Create auto-drafts for each theme template files.
+	$block_template_files = gutenberg_get_template_paths();
```

Comment:
As already mentioned, accessing these files on each request is going to have a negative impact on performance. We can probably store this info once on theme activation and we'll also need it after theme updates.